e2e/config/config.go

```diff
@@ -195,6 +280,29 @@ func PVCSpecsMap(config *types.Config) map[string]types.PVCSpecConfig {
 	return res
 }
 
+// validateK8sResourceName validates a Kubernetes resource name according to RFC 1123 subdomain format.
+// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names.
+// It ensures that the resource name:
+// 1. Has a length no greater than 253 characters.
+// 2. Contains only lowercase alphanumeric characters, '-' or '.'.
+// 3. Starts and ends with an alphanumeric character.
+// Returns an error if the name does not meet these criteria.
+func validateK8sResourceName(resourceName string) error {
+	const maxLen = 253
```

Comment:
limit is actually 253 chars for subdomain names:
```
kubectl create -f sc2.yaml --context hub
The StorageClass "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" is invalid: metadata.name: Invalid value: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa": must be no more than 253 characters
```

Reply:
Let’s use k8s.io/apimachinery/pkg/api/validation